### PR TITLE
[NOT FOR MERGE] Add Blank Skip to Zipformer2

### PR DESCRIPTION
**Pull request by ezerhouni**
I will double check that I am not missing something.

cc @csukuangfj

### CODE REVIEW

**Comment by yfyeung** [comment]
Thanks!

File: egs/librispeech/ASR/zipformer/lconv.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+from typing import Optional
+
+import torch
+import torch.nn as nn
+from acoustic_model.utils_py.scaling_zipformer import Balancer, ScaledConv1d
+
+
+class LConv(nn.Module):
+    """A convolution module to prevent information loss."""
```
Comment (danpovey) [inline egs/librispeech/ASR/zipformer/lconv.py:25]: should probably clarify that this relates to blank skipping and maybe reference the paper.

File: egs/librispeech/ASR/zipformer/train.py [hunk 2]
```diff
@@ -620,16 +640,24 @@ def get_model(params: AttributeDict) -> nn.Module:
...
         decoder=decoder,
         joiner=joiner,
+        lconv=lconv,
+        frame_reducer=frame_reducer,
         encoder_dim=max(_to_int_tuple(params.encoder_dim)),
         decoder_dim=params.decoder_dim,
         vocab_size=params.vocab_size,
         use_transducer=params.use_transducer,
         use_ctc=params.use_ctc,
+        use_bs=params.use_bs,
```
Comment (danpovey) [inline egs/librispeech/ASR/zipformer/train.py:660]: I think it would be better to call this use_blank_skip.

**Review by danpovey — COMMENTED** [review]
A couple comments (not necessarily promising to merge this, must discuss with the guys.)

Also: since you are only comparing warmup with 2.0, and warmup is now not used elsewhere, it would be better to have a config "blank_skip_start_batch" and just pass in a bool that says whether we do blank slipping.

What happens in test time?  And I wonder whether this would require changes to sherpa.

The guys would have to comment before we were  to merge this, my comments can be ignored if we are not going to merge.

**Review by yaozengwei — COMMENTED** [review]
I remember that @yfyeung said removing the Lconv module gets better result.

**Comment by ezerhouni** [comment]
@csukuangfj @danpovey If merging this PR is not very urgent and useful (>2 weeks), I can run it librispeech with all the changes needed (from review this PR) update Results.md

**Comment by danpovey** [comment]
We also need to see whether this might cause a problem for streaming systems, e.g. does it support streaming_forward().

**Comment by csukuangfj** [comment]
> @csukuangfj @danpovey If merging this PR is not very urgent and useful (>2 weeks), I can run it librispeech with all the changes needed (from review this PR) update Results.md

That would be great. We have not used blank skipping with the latest zipformer.

**Comment by ezerhouni** [comment]
@danpovey It should work with streaming_forward 
@csukuangfj I will try to schedule that asap. In the meantime, any review would be welcome (so that I can test the reviewed code)

Thank you
